### ISSUE TITLE
Option to set umask of exec-user

### ISSUE THREAD

**Issue by leberknecht** [opencontainers/runc#1650]
Hi, 
As mentioned here https://github.com/moby/moby/issues/19189 it would be nice to have an option to enforce umask setting of the executing user.

**Comment by cyphar** [comment]
So, this would need to be done through the `config.json` for OCI runtimes (which requires a specification. change), As far as I can tell, you're not really familiar with OCI specifications, so I could submit the relevant PR for you.

However, there is a property of `umask(2)` that might make this unnecessary. Most notably, `umask(2)` is preserved across both `fork(2)` and `execve(2)` -- so really whatever is calling `runc` is able to set its own `umask(2)` explicitly. In the world of Docker, this would mean that containerd could set `umask(2)` before spawning `runc`.

**Comment by cyphar** [comment]
I'm reopening, since I didn't mean to say that this is something that `runc` shouldn't do -- I was just mentioning that as a point that might come up in the spec discussion. :wink:

**Comment by cyphar** [comment]
I've opened the OCI spec PR here: opencontainers/runtime-spec#941.

**Comment by phod** [comment]
Has there been any progress on this issue recently? Anything that needs unblocking? Thanks.

**Comment by cyphar** [comment]
The OCI spec PR needs to be merged first.

**Comment by leberknecht** [comment]
@cyphar meanwhile the PR was merged :) what would be the next steps? Anything we can help with?

**Comment by leberknecht** [comment]
@cyphar ping

**Comment by cyphar** [comment]
Sorry I missed your message from a year ago :grimacing:. I'm pretty sure this is fully supported since #2527 -- which will be in the next release. So the next steps would be to add containerd and Docker support for it, and then pull in the new version of runc which I'll be releasing tomorrow.

**Comment by ajschmidt8** [comment]
@cyphar, just wanted to follow up here as I'm interested in this issue as well. How can I tell if `containerd` and `Docker` currently support this now?

**Comment by leberknecht** [comment]
On a side-note: meanwhile the `umask` support has arrived in podman:
```
% podman run alpine /bin/sh -c 'umask'
0022
% podman run --umask 002 alpine /bin/sh -c 'umask'
0002
```

**Comment by kolyshkin** [comment]
This feature was implemented in https://github.com/opencontainers/runc/pull/2527 and is available since runc v1.0.0-rc93, released in Feb 2021.

Guess we can close this one now.
